lib/matplotlib/_constrained_layout.py

```diff
@@ -561,32 +525,21 @@ def layoutcolorbarsingle(ax, cax, shrink, aspect, location, pad=0.05):
 
 def _getmaxminrowcolumn(axs):
     # helper to get the min/max rows and columns of a list of axes.
-    maxrow = -100000
-    minrow = 1000000
-    maxax = None
-    minax = None
-    maxcol = -100000
-    mincol = 1000000
-    maxax_col = None
-    minax_col = None
 
-    for ax in axs:
-        subspec = ax.get_subplotspec()
-        nrows, ncols, row_start, row_stop, col_start, col_stop = \
-            subspec.get_rows_columns()
-        if row_stop > maxrow:
-            maxrow = row_stop
-            maxax = ax
-        if row_start < minrow:
-            minrow = row_start
-            minax = ax
-        if col_stop > maxcol:
-            maxcol = col_stop
-            maxax_col = ax
-        if col_start < mincol:
-            mincol = col_start
-            minax_col = ax
-    return (minrow, maxrow, minax, maxax, mincol, maxcol, minax_col, maxax_col)
+    def key(pos_ax):
+        pos, ax = pos_ax
+        return pos
+
+    minrow, minrow_ax = min(
```

Comment:
This code block is pretty obfuscated compared to the original.

Reply:
fair enough, went for some compromise...